src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -453,7 +453,8 @@ private String DecodeEntryName(Byte[] entryNameBytes)
                 readEntryNameEncoding = Encoding.UTF8;
             }
 
-            return new string(readEntryNameEncoding.GetChars(entryNameBytes)); //readEntryNameEncoding.GetString(entryNameBytes);
+            fixed (byte* bytePtr = entryNameBytes)
+                return readEntryNameEncoding.GetString(bytePtr, entryNameBytes.Length);
```

Comment:
Why are you using unsafe code here?  It's good to make a change to avoid the intermediate chars[], but why not just do ```return readEntryNameEncoding.GetString(entryNameBytes);``` as was in the commented out code?